openwisp_users/tests/test_models.py

```diff
@@ -181,3 +183,22 @@ def test_add_users_with_empty_phone_numbers(self):
         self.assertIsNone(user1.phone_number)
         self.assertIsNone(user2.phone_number)
         self.assertEqual(self.user_model.objects.filter(phone_number=None).count(), 2)
+
+    def test_cache_user_permission(self):
+        user = self.user_model(
+            username='user', email='email1@email.com', password='user1',
+        )
+        user.full_clean()
+        user.save()
+        user.refresh_from_db()
+        group = Group.objects.filter(name='Operator')
+        user.groups.set(group)
+
+        self.assertEqual(user.get_all_permissions(), user.get_permissions)
+
+        user.groups.set(Group.objects.filter(name='Administrator'))
+        self.assertEqual(user.get_all_permissions(), user.get_permissions)
+
+        permission = Permission.objects.filter(codename='add_organization')
+        user.user_permissions.add(*permission)
+        self.assertEqual(user.get_all_permissions(), user.get_permissions)
```

Comment:
please use assertNumQueries to ensure the query is executed only at the first call